### ISSUE TITLE
Remove profiles

### ISSUE THREAD

**Issue by fwinkelbauer** [fwinkelbauer/Bumpy#33]
So I've been thinking. The reason I introduced profiles was to limit an operation to a group of versions. This was needed because a version such as `1.0.0` cannot be changed by `bumpy increment 4`. Also some versions cannot deal with the semver format, meaning that an operation such as `bumpy label -alpha` cannot be applied to all versions. Bumpy handles these cases by throwing an error, which stops the current operation.

Could we get rid of profiles if we would simply keep going after an error?

**Example:**

`bumpy increment 4`

Output:

```
some_file.txt (4): 1.0.0 -> cannot apply operation
other_file.txt (1): 1.0.0.0 -> 1.0.0.1
```

@euronay Do you have any opinion/suggestion regarding this ticket?

**Comment by fwinkelbauer** [comment]
To give a little more context: My main goal is to simplify a basic `.bumpyconfig` so that it could look like this in the future:

```
MyProject\**\AssemblyInfo.cs
MyNuSpec\**\*.nuspec
```

The potential problem with such an approach is that an AssemblyInfo can contain two different kinds of versions:

- `AssemblyVersion` and `AssemblyFileVersion` share the format `1.0.0.0`
- `AssemblyInformationalVersion` is less strict, so something like `1.0.0-alpha` is ok

My first thought was that I would need to remove profiles to solve this problem in a "clean" way, but after a good nights sleep I am now aware that the purpose of templates is to cover the simplest case. So If anyone would need two different types of versions in an AssemblyInfo file, they could just adapt the configuration to their needs.

**TL;DR** I think that profiles should stay, but I am still unsure if implementing a "continue on error" feature would be a good idea.

**Comment by euronay** [comment]
I can see where you are coming from, but having the format string in the profile is actually a useful tool.  In one of the projects I'm working on, I have some old setup projects with content like this:

```
"AspNetVersion" = "8:4.0.30319.0"
"ProductVersion" = "8:02.09.0013"
```

With the profiles the way they are, I can use the regex to my advantage and only update the version number I am interested by using this in the `.bumpyconfig`:

`*.vdproj = "ProductVersion" = "8\:(?<version>\d+\.\d+\.\d+)`

A "continue on error" feature might be handy - but I think it would have to be an option. I'm using bumpy in cake - so if a version number is not updated, I want it to throw an error.

Appreciate the good work! Cheers

**Comment by fwinkelbauer** [comment]
Thanks for the insight!

I think I'll try this: Bumpy will continue if it encounters an error, but it will report it via console/exit code. This would make sense as it should not stop arbitrary depending on the configuration (in my opinion). In the future I could add a flag which could change this behaviour (abort, continue, continue & ignore).
